src/connection.cpp

```diff
@@ -64,7 +67,14 @@ namespace oxen::quic
         return 0;
     }
 
-    int recv_stream_data(
+    int on_recv_datagram(ngtcp2_conn* /* conn */, uint32_t flags, const uint8_t* data, size_t datalen, void* user_data)
+    {
+        log::trace(log_cat, "{} called", __PRETTY_FUNCTION__);
+        return static_cast<Connection*>(user_data)->recv_datagram(
+                {reinterpret_cast<const std::byte*>(data), datalen}, flags & NGTCP2_STREAM_DATA_FLAG_FIN);
```

Comment:
Receive datagram has no `FIN` flag.  It seems to only have one flag, `0RTT`, which we don't support so don't care about.

The recv_datagram function below need not take a `bool fin` for the same reason.